salt/modules/boto_route53.py

```diff
@@ -340,6 +341,107 @@ def create_zone(zone, private=False, vpc_id=None, vpc_region=None, region=None,
     return True
 
 
+def create_healthcheck(ip_addr=None, fqdn=None, region=None, key=None, keyid=None, profile=None,
+                      port=53, hc_type='TCP', resource_path='', string_match=None, request_interval=30,
+                      failure_threshold=3, retry_on_errors=True, error_retries=5):
+    '''
+    Create a Route53 healthcheck
+
+    .. versionadded:: Oxygen
+
+    ip_addr
+
+        ip address to check.  ip_addr or fqdn is required.
+
+    fqdn
+
+        domain name of the endpoint to check.  ip_addr or fqdn is required
+
+    port
+
+        port to check
+
+    hc_type
+
+        Healthcheck type.  HTTP | HTTPS | HTTP_STR_MATCH | HTTPS_STR_MATCH | TCP
+
+    resource_path
+
+        Path to check
+
+    string_match
+
+        if hc_type is HTTP_STR_MATCH or HTTPS_STR_MATCH, the string to search for in the
+        response body from the specified resource
+
+    request_interval
+
+        The number of seconds between the time that Amazon Route 53 gets a response from
+        your endpoint and the time that it sends the next health-check request.
+
+    failure_threshold
+
+        The number of consecutive health checks that an endpoint must pass or fail for
+        Amazon Route 53 to change the current status of the endpoint from unhealthy to
+        healthy or vice versa.
+
+    region
+
+        region endpoint to connect to
+
+    key
+
+        AWS key
+
+    keyid
+
+        AWS keyid
+
+    profile
+
+        AWS pillar profile
+
+    CLI Example::
+
+        salt myminion boto_route53.create_healthcheck 192.168.0.1
+        salt myminion boto_route53.create_healthcheck 192.168.0.1 port=443 hc_type=HTTPS \
+                                                      resource_path=/ fqdn=blog.saltstack.furniture
+    '''
+    if fqdn is None and ip_addr is None:
+        log.error('One of the following must be specified: fqdn or ip_addr')
+        return None
```

Comment:
I know the rest of the module returns None/True/False, but we're trying to move to a return format that always returns a dict, similar to:

```
return {'results': my_return}
```

Or in the case of failure:

```
return {'error': error_to_return}
```

This lets us cleanly return errors back to end-users through the state modules in cases of unexpected failure without throwing stacktraces or raising exceptions across module boundaries.

As an example:

https://github.com/saltstack/salt/blob/develop/salt/modules/boto_sqs.py#L104-L123

https://github.com/saltstack/salt/blob/develop/salt/states/boto_sqs.py#L118-L128

Reply:
:+1: updated